packages/lisk-transactions/src/utils/validation/validate_transaction.js

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright © 2018 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ *
+ */
+
+import validator from './validator';
+import * as schemas from './schema';
+
+const schemaMap = {
+	0: validator.compile(schemas.transferTransaction),
+	1: validator.compile(schemas.signatureTransaction),
+	2: validator.compile(schemas.delegateTransaction),
+	3: validator.compile(schemas.voteTransaction),
+	4: validator.compile(schemas.multiTransaction),
+	5: validator.compile(schemas.dappTransaction),
+};
+
+const getTransactionValidator = type => {
+	const schema = schemaMap[type];
+	if (!schema) {
+		throw new Error('Unsupported transaction type.');
+	}
+	return schema;
+};
+
+const validateMultiTransaction = tx => {
+	if (tx.asset.multisignature.min > tx.asset.multisignature.keysgroup.length) {
+		return {
+			valid: false,
+			errors: [
+				{
+					dataPath: '.asset.multisignature.min',
+					message:
+						'.asset.multisignature.min cannot be greater than the keysgroup',
+				},
+			],
+		};
+	}
+	return {
+		valid: true,
+		errors: null,
+	};
+};
+
+const validateTransaction = tx => {
+	const validate = getTransactionValidator(tx.type);
+	const valid = validate(tx);
+	// Ajv produces merge error when error happens within $merge
```

Comment:
We don't have to, but it creates duplicate error (one for merge and one for actual error).
So I think it would be better to filter out error which is not really useful